mender/templates/api-gateway/hpa.yaml

```
@@ -0,0 +1,32 @@
+{{- if .Values.api_gateway.hpa.enabled }}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ include "mender.fullname" . }}-api-gateway
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ include "mender.fullname" . }}-api-gateway
+  minReplicas: {{ .Values.api_gateway.replicas }}
```

If HPA is enabled, the `spec.replicas` for deployments should not be set, otherwise you'll get a temporary change of replicas whenever the deployment resource is created.
```suggestion
  minReplicas: {{ .Values.api_gateway.minReplicas }}
```